src/Symfony/Component/Mailer/Transport.php

```diff
@@ -0,0 +1,180 @@
+                }
+
+                if ('smtp' === $parsedDsn['scheme']) {
+                    return new Sendgrid\Smtp\SendgridTransport($user, $dispatcher, $logger);
+                }
+                if ('api' === $parsedDsn['scheme']) {
+                    return new Sendgrid\Http\Api\SendgridTransport($user, $client, $dispatcher, $logger);
+                }
+
+                throw new LogicException(sprintf('The "%s" scheme is not supported for mailer "%s".', $parsedDsn['scheme'], $parsedDsn['host']));
+            case 'ses':
+                if (!class_exists(Amazon\Smtp\SesTransport::class)) {
+                    throw new \LogicException('Unable to send emails via Amazon SES as the bridge is not installed. Try running "composer require symfony/amazon-bridge".');
+                }
+
+                if ('smtp' === $parsedDsn['scheme']) {
+                    return new Amazon\Smtp\SesTransport($user, $pass, $query['region'] ?? null, $dispatcher, $logger);
+                }
+                if ('api' === $parsedDsn['scheme']) {
+                    return new Amazon\Http\Api\SesTransport($user, $pass, $query['region'] ?? null, $client, $dispatcher, $logger);
+                }
+                if ('http' === $parsedDsn['scheme']) {
+                    return new Amazon\Http\SesTransport($user, $pass, $query['region'] ?? null, $client, $dispatcher, $logger);
+                }
+
+                throw new LogicException(sprintf('The "%s" scheme is not supported for mailer "%s".', $parsedDsn['scheme'], $parsedDsn['host']));
+            case 'mandrill':
+                if (!class_exists(Mailchimp\Smtp\MandrillTransport::class)) {
+                    throw new \LogicException('Unable to send emails via Mandrill as the bridge is not installed. Try running "composer require symfony/mailchimp-bridge".');
+                }
+
+                if ('smtp' === $parsedDsn['scheme']) {
+                    return new Mailchimp\Smtp\MandrillTransport($user, $pass, $dispatcher, $logger);
+                }
+                if ('api' === $parsedDsn['scheme']) {
+                    return new Mailchimp\Http\Api\MandrillTransport($user, $client, $dispatcher, $logger);
+                }
+                if ('http' === $parsedDsn['scheme']) {
+                    return new Mailchimp\Http\MandrillTransport($user, $client, $dispatcher, $logger);
+                }
+
+                throw new LogicException(sprintf('The "%s" scheme is not supported for mailer "%s".', $parsedDsn['scheme'], $parsedDsn['host']));
+            default:
+                if ('smtp' === $parsedDsn['scheme']) {
+                    return new Transport\Smtp\EsmtpTransport($parsedDsn['host'], $parsedDsn['port'] ?? 25, $query['encryption'] ?? null, $query['auth_mode'] ?? null, $dispatcher, $logger);
+                }
+
+                throw new LogicException(sprintf('The "%s" mailer is not supported.', $parsedDsn['host']));
```

Comment:
It probably would be `\LogicException`

Reply:
no

Reply:
For clarity, Symfony defines it own exception classes (based of SPL exceptions) which implement the components `ExceptionInterface` marker interface (no methods) to make it easier to catch exception only related to a specific component.